openedx/core/djangoapps/user_api/management/commands/populate_enterprise_learner_current_job.py

```diff
@@ -0,0 +1,150 @@
+
+    Example usage:
+        $ # Update the current job of learners.
+        $ ./manage.py lms populate_enterprise_learner_current_job
+    """
+
+    help = 'Populates current job of learners in extended learner profile.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--oauth-host', required=False,
+                            default=settings.ENTERPRISE_BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL)
+        parser.add_argument('--client-id', required=False,
+                            default=settings.ENTERPRISE_BACKEND_SERVICE_EDX_OAUTH2_KEY)
+        parser.add_argument('--client-secret', required=False,
+                            default=settings.ENTERPRISE_BACKEND_SERVICE_EDX_OAUTH2_SECRET)
+
+    def _get_edx_api_client(self, config):
+        return OAuthAPIClient(
+            base_url=config['oauth_host'],
+            client_id=config['client_id'],
+            client_secret=config['client_secret'],
+        )
+
+    def _fetch_learner_job_data(self, config):
+        """
+        Get the username and current job of learners from discovery service.
+
+        Returns:
+            list: List of dictionaries containing username and current job of learners.
+        """
+        try:
+            client = self._get_edx_api_client(config)
+
+            course_discovery_url = settings.COURSE_CATALOG_URL_ROOT
+            learner_jobs_endpoint = '/taxonomy/api/v1/learners-current-job/'
+            url = urljoin(course_discovery_url, learner_jobs_endpoint)
+
+            response = client.get(url)
+            response.raise_for_status()
+
+            return response.json() if response else []
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception('Error while fetching username and current_job data.')
+            raise
+
+    def _get_user_profile(self, username):
+        """
+        Helper method to return the user profile object based on username.
+        """
+        try:
+            existing_user = User.objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise errors.UserNotFound()  # lint-amnesty, pylint: disable=raise-missing-from
+
+        existing_user_profile = UserProfile.objects.get(user=existing_user)
+
+        return existing_user_profile
+
+    def _update_current_job_of_learner(self, user_profile, current_job):
+        """
+        Update the current job of learner in their extended profile.
+
+        Args:
+            user_profile (UserProfile): Extended profile of the learner.
+            current_job (number): Current job of the learner.
+        """
+        try:
+            meta = user_profile.get_meta()
+            meta['enterprise_learner_current_job'] = current_job
+            user_profile.set_meta(meta)
+            user_profile.save()
+        except Exception:
+            LOGGER.exception(f'Error while updating current job of {user_profile.user.username}.')
+            raise
```

Comment:
I am not sure if this log will serve any purpose? We are re-raising the exception and it will print the whole stack trace anyway.